src/server/app.ts

```diff
@@ -48,6 +51,27 @@ function addRoutes(attach: string, router: Router | Handler): void {
   app.use(SERVER_SETTINGS.getServerRoot() + attach, router);
 }
 
+// Add request metrics
+app.use((req: Request, res: Response, done) => {
+  const stopRequestTimer = metrics.requestDuration.startTimer();
+
+  function addRequestMetrics() {
+    const lables = { host: req.hostname, route: req.path, method: req.method, status: res.statusCode };
+    res.removeListener("finish", addRequestMetrics);
+    res.removeListener("close", addRequestMetrics);
+
+    metrics.totalRequests.inc(lables);
+    stopRequestTimer(lables);
+  }
+
+  res.on("finish", addRequestMetrics);
```

Comment:
It seems using [`on-finished`](https://github.com/jshttp/on-finished) is now the recommended way to do this. We then shouldn't need to use any other `on` or `removeListener` functions.